### ISSUE TITLE
Add reconnect to server button

### ISSUE THREAD

**Issue by Zowdie** [treeform/istrolid-bugs#19]
Sometimes if you're in a game, and the game freezes, it becomes impossible to reconnect to the server if it is already locked, resulting in a player being unable to do anything and unable to reconnect without opening the console. Adding a button right above the "Leave Game" one that runs `battleMode.joinServer(battleMode.serverName);` would solve that problem.

**Comment by treeform** [comment]
You mean its locked if there are too many people on a server?

**Comment by Zowdie** [comment]
Yes. In rare cases your game will freeze while you're in a match, and there is no convenient way without scripting to get back in to un-freeze the game if there are already too many people in the server.

**Comment by treeform** [comment]
Yeah that is stupid. I will think about this. 

Do you know why guy freezes in the first place?
Does it freeze in like you can't click on anything?
Does it freeze in like the game still works but it feels disconnected?
Can you open up a console (pressing F10) and see if it has any errors?

**Comment by Zowdie** [comment]
Freezing seems pretty arbitrary but sometimes happens when people leave the game. You CAN click on things but the game itself freezes (the ships, projectiles, etc). No ships work, and yeah it does feel disconnected.

**Comment by lectrician1** [comment]
Also if you become disconnected say if your wifi goes down and you want to rejoin the battle, the moment you login, you should be redirected back to the battle. This way it makes sure you don't have to find the server again and you can get back to your needy teammates faster. This also has to do with another issue (see #36 ).

**Comment by Zowdie** [comment]
I am not referring to that type of disconnection; freezing does not disconnect you, rather it is a problem with the client freezing up. Unrelated to that issue.

**Comment by apatarty** [comment]
it happens in moments of extreme lag for me

**Comment by lectrician1** [comment]
Also, the reconnect rate should be faster. I usually play Istrolid on a laptop (browser version) and when I close it for a period of time and then reopen it, it can take up to 30 seconds for the battle to re-sync (the reconnect to server button then pops up). I also think there shouldn't be a reconnect to server button at all. Have it so that the player remains in an AFK state until the browser/steam window is refocused. Then add a "Reconnecting..." alert stating that the game is trying to connect.

**Comment by Vaxyyy** [comment]
This is a little addon that can fix this problem, it adds a button to the top right that will force disconnect form server you are on and then re-join it with the suggested way.

- [reconnectButton code](https://github.com/Vaxyyy/Istrolid-Scripts/blob/master/modules/reconnectButton.js)
